### ISSUE TITLE
inject returning wrong type when using treatDefaultAsFactory

### ISSUE THREAD

**Issue by johannes-z** [vuejs/composition-api#866]
The type returned from `inject` resolves to the factory function, rather than the object returned from the factory. For example:

```
const value = inject('key', () => 1, true)
```

This type of `value` is `() => 1` instead of `1`.

The typescript signature for the function should be: 

```ts
declare function inject<T>(key: InjectionKey<T> | string, defaultValue: T | (() => T), treatDefaultAsFactory?: boolean): T;
                                                                         ^^^^^^^^^^^^
```

I changed this in `vue-composition-api.d.ts` and it works as expected.

**Comment by MinatoHikari** [comment]
I have found this difference when coding computed-inject for vueuse, let me fix it.

**Comment by xiaoxiangmoe** [comment]
fixed in v1.4.4
